### PR TITLE
 A section - finished

### PR DESCRIPTION
**Pull request by tsoriLache**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,9 +19,13 @@
   },
   "homepage": "https://github.com/noamgolani/phonebookDB#readme",
   "dependencies": {
-    "express": "^4.17.1"
+    "express": "^4.17.1",
+    "morgan": "^1.10.0",
```
Comment (noamgolani) [inline package.json]: should be dev dependency

File: server/app.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const express = require("express");
+const app = express();
+const personsRouter = require('./routers/persons_router')
```
Comment (noamgolani) [inline server/app.js]: persons_route should be names personsRoute

File: server/app.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+morgan.token('body', req => {
+    return JSON.stringify(req.body)
+  })
+  
+app.use(morgan(':method :url :status - :response-time ms :body'))
+
+app.use('/api/persons',personsRouter)
+
+app.get('/info', (request, response) => {
```
Comment (noamgolani) [inline server/app.js]: can move into diferent file

File: server/helpers/validation.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+function isNumberValid(number){
+    const arrNumber = number.split("");
+    arrNumber.splice(3,1);
```
Comment (noamgolani) [inline server/helpers/validation.js]: Why is this here? maby add a comment

File: server/helpers/validation.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    return( number.length===11 &&
+            number.slice(3,4)==='-' && 
+            arrNumber.every(digit => digit>=0&&digit<=9) &&
```
Comment (noamgolani) [inline server/helpers/validation.js:6]: if its a digit it cant be  out of those bounds

File: server/lib/dummyDB.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
+const { search } = require("superagent");
+const { v4: uuidv4 } = require('uuid');
+
+let contacts = []
```
Comment (noamgolani) [inline server/lib/dummyDB.js:4]: Allthe function here should be async, to replace a real db request

File: server/routers/persons_router.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+const {getAll,getSingle,deleteSingle,addContact,isNameExist} = require('../lib/dummyDB');
+const {isNumberValid} = require('../helpers/validation');
+
+
+router.get('/', (request, response) => {
+    response.json(getAll())
+})
+
+router.get('/:id', (request, response) => {
+    const id = Number(request.params.id);
```
Comment (noamgolani) [inline server/routers/persons_router.js]: can use `+request.params.id` 
instead

File: server/routers/persons_router.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    if(getSingle(id)){
```
Comment (noamgolani) [inline server/routers/persons_router.js]: why fetch the getSingle twice

File: server/routers/persons_router.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        response.status(204).end()
+    }else{
+        response.status(400);
+        response.send('id is not found');
+    }
+})
+
+router.post('/', (request, response) => {
+    const {name,number} = request.body;
+    if(name&&number){   //add number validation
```
Comment (noamgolani) [inline server/routers/persons_router.js]: add number validator

File: server/helpers/validation.js (hunk 5 above)
Comment (tsoriLache) [reply server/helpers/validation.js:6]: didn't understand.. digit are usuly between 0-9

File: server/routers/persons_router.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    }else{
+        response.status(400);
+        response.send('id is not found');
+    }
+})
+
+router.delete('/:id', (request, response) => {
+    const id = Number(request.params.id);
+    if(getSingle(id)){
+        deleteSingle(id)
```
Comment (noamgolani) [inline server/routers/persons_router.js]: make it so that delete will throw an error if bad id.

**Review by noamgolani — COMMENTED** [review]
nice
